drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -6082,7 +6085,12 @@ Vector<uint8_t> RenderingDeviceVulkan::buffer_get_data(RID p_buffer, uint32_t p_
 	}
 
 	Buffer tmp_buffer;
-	_buffer_allocate(&tmp_buffer, p_size, VK_BUFFER_USAGE_TRANSFER_DST_BIT, VMA_MEMORY_USAGE_AUTO_PREFER_HOST, VMA_ALLOCATION_CREATE_HOST_ACCESS_RANDOM_BIT);
+#ifdef __ANDROID__
+	bool p_coherent = true;
```

Comment:
It seems odd to me that this is only being used on a single platform when the API is supposed to be platform-independent. If this is needed for a particular hardware configuration shouldn't we be testing for that configuration instead of just testing for the platform? If there is no way to test for the configuration then I think there should be documentation here to indicate when this flag needs to be set.

Reply:
`VK_MEMORY_PROPERTY_HOST_COHERENT_BIT`  MacOS would have no problem with this, but I have not checked the performance in detail.

**on Android:**
`Vulkan API 1.1.128 - Forward Mobile - Using Vulkan Device #0: Qualcomm - Adreno (TM) 650`

**on MacOS:**
`Vulkan API 1.2.231 - Forward Mobile - Using Vulkan Device #0: Apple - Apple M1`

------------------------
@sakrel has posted this information, who can test all this?
https://asawicki.info/news_1740_vulkan_memory_types_on_pc_and_how_to_use_them

1. The Intel way
2. The NVIDIA way
3. The AMD way
4. The SAM way
5. The APU way

Reply:
**"Coherent Memory"** could perhaps be useful for testing on more hardware `Rendering -> Rendering Device`? "Staging Buffer" is the other function that also uses `VMA_MEMORY_USAGE_AUTO_PREFER_HOST`. I have not yet checked whether the function can also cause problems on Android.

![image](https://github.com/godotengine/godot/assets/41921395/3fd54cfa-23c0-4d60-9f03-1afcc2b6240e)